### PR TITLE
feat(ac): add register page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add ac register page

image

## Linear Issue Reference

log-2407

## Testing

test locally with mock hasAdminUser

@logto-io/eng

### CODE REVIEW

**Comment by wangsijie** [comment]
I remember you are going to move the register page to `UI`, just want to make sure that this is the final plan.

File: packages/phrases/src/locales/en.ts [hunk 1]
```diff
@@ -547,6 +547,29 @@ const translation = {
...
+        password: 'Password',
+        confirm_password: 'Confirm password',
+      },
+      description: {
+        welcome_title: 'Welcome to Logto Admin Console',
+        welcome_description:
+          'Admin console is a web app to manage Logto without coding requirements. Let’s first create an admin account.',
+      },
+      error: {
+        username_required: 'Username is required',
```
Comment (IceHe) [inline packages/phrases/src/locales/en.ts:565]: nit: `Require username` better?

File: packages/phrases/src/locales/en.ts [hunk 2]
```diff
@@ -547,6 +547,29 @@ const translation = {
...
+        username_invalid_format:
+          'Username Should not start with a number, only  letters, numbers and underscores is allowed.',
```
Comment (IceHe) [inline packages/phrases/src/locales/en.ts:567]: - `Should` → `should`
- two whitespaces → one whitespace

File: packages/phrases/src/locales/en.ts [hunk 3]
```diff
@@ -547,6 +547,29 @@ const translation = {
...
+        password_min_length: 'Password should be at least {{min}} characters',
+        password_required: 'Password is required',
```
Comment (IceHe) [inline packages/phrases/src/locales/en.ts:569]: nit: ditto: `Require password` better?

**Comment by demonzoo** [comment]
Aren't these 2 png files too big? 46KB for each is way bigger than svg format

File: packages/console/src/icons/ClearIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      viewBox="0 0 24 24"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      onMouseDown={onMouseDown}
+    >
+      <path
+        fillRule="evenodd"
+        clipRule="evenodd"
+        d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21ZM8.46447 9.87868C8.07394 9.48816 8.07394 8.85499 8.46447 8.46447C8.85499 8.07394 9.48816 8.07394 9.87868 8.46447L12 10.5858L14.1213 8.46447C14.5118 8.07394 15.145 8.07394 15.5355 8.46447C15.9261 8.85499 15.9261 9.48815 15.5355 9.87868L13.4142 12L15.5355 14.1213C15.9261 14.5118 15.9261 15.145 15.5355 15.5355C15.145 15.9261 14.5118 15.9261 14.1213 15.5355L12 13.4142L9.87868 15.5355C9.48816 15.9261 8.85499 15.9261 8.46447 15.5355C8.07394 15.145 8.07394 14.5118 8.46447 14.1213L10.5858 12L8.46447 9.87868Z"
+        fill="currentcolor"
```
Comment (demonzoo) [inline packages/console/src/icons/ClearIcon.tsx:16]: `currentColor`

File: packages/console/src/pages/Register/Input/index.module.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  transition-property: outline, border;
+  transition-timing-function: ease-in-out;
+  transition-duration: 0.2s;
+  font: var(--font-body-medium);
+  display: flex;
+  align-items: center;
+  background-color: var(--color-base);
+  overflow: hidden;
+
+
```
Comment (demonzoo) [inline packages/console/src/pages/Register/Input/index.module.scss:16]: extra empty line?

File: packages/console/src/pages/Register/Input/index.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import ClearIcon from '@/icons/ClearIcon';
+
+import * as styles from './index.module.scss';
+
+type Props = HTMLProps<HTMLInputElement> & {
+  error?: string;
+  className?: string;
+  showClearButton?: boolean;
+  onClear?: () => void;
+  errorStyling?: boolean;
```
Comment (demonzoo) [inline packages/console/src/pages/Register/Input/index.tsx:13]: What does this do?

File: packages/console/src/pages/Register/Input/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  }: Props,
+  reference: ForwardedRef<HTMLInputElement>
+) => {
+  return (
+    <div className={className}>
+      <div className={classNames(styles.container, error && errorStyling && styles.error)}>
+        <input {...rest} ref={reference} />
+        {showClearButton && onClear && (
+          <ClearIcon
+            className={styles.actionButton}
```
Comment (demonzoo) [inline packages/console/src/pages/Register/Input/index.tsx:34]: Should we have some hover effect on this clear button?

File: packages/console/src/pages/Register/index.module.scss [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+.placeholder {
+  flex: 1;
+}
+
+
```
Comment (demonzoo) [inline packages/console/src/pages/Register/index.module.scss:16]: extra empty line?

File: packages/phrases/src/locales/en.ts (hunk 2 above)
Comment (demonzoo) [reply packages/phrases/src/locales/en.ts:567]: `are` allowed

**Comment by simeng-li** [comment]
wait for new design

**Pull request closed without merging**